### PR TITLE
feat(argocd-apps): Support Template Patch to ApplicationSet

### PR DESCRIPTION
**Pull request by yu-croco**
Resolves https://github.com/argoproj/argo-helm/issues/2463 

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argocd-apps/ci/applicationsets-multiple-sources-values.yaml [hunk 1]
```diff
@@ -22,3 +22,16 @@ applicationsets:
       destination:
         server: '{{url}}'
         namespace: default
+  templatePatch: |
```
Comment (yu-croco) [inline charts/argocd-apps/ci/applicationsets-multiple-sources-values.yaml:25]: It's from https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Template/#template-patch

File: charts/argocd-apps/templates/applicationsets.yaml [hunk 2]
```diff
@@ -84,4 +84,8 @@ spec:
       {{- end }}
     {{- end -}}
   {{- end }}
+  {{- with .templatePatch }}
+  templatePatch: |
+    {{- . | nindent 4 }}
+  {{- end }}
```
Comment (yu-croco) [inline charts/argocd-apps/templates/applicationsets.yaml:90]: ✏️  https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Template/#template-patch

**Review by mkilchhofer — APPROVED** [review]
LGTM


<details>

<summary>Rendered output</summary>

```bash
$ helm template . --values ci/applicationsets-values.yaml
---
# Source: argocd-apps/templates/applicationsets.yaml
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: applicationset
spec:
  generators:
    - git:
        directories:
        - path: guestbook
        - path: kustomize-*
        repoURL: https://github.com/argoproj/argocd-example-apps.git
        revision: HEAD
  ignoreApplicationDifferences:
    - jsonPointers:
      - /spec/syncPolicy
  syncPolicy:
    preserveResourcesOnDeletion: false
  template:
    metadata:
      name: '{{path.basename}}'
    spec:
      project: 'default'
      source:
        path: '{{path}}'
        repoURL: https://github.com/argoproj/argocd-example-apps.git
        targetRevision: HEAD
      destination:
        namespace: default
        server: https://kubernetes.default.svc
      syncPolicy:
        automated:
          prune: false
          selfHeal: false
      ignoreDifferences:
        - group: apps
          jsonPointers:
          - /spec/replicas
          kind: Deployment
      info:
        - name: url
          value: https://argoproj.github.io/
  templatePatch: |
    spec:
      source:
        helm:
          valueFiles:
          {{- range $valueFile := .valueFiles }}
            - {{ $valueFile }}
          {{- end }}
    {{- if .autoSync }}
      syncPolicy:
        automated:
          prune: {{ .prune }}
    {{- end }}
---
# Source: argocd-apps/templates/applicationsets.yaml
apiVersion: argoproj.io/v1alpha1
kind: ApplicationSet
metadata:
  name: applicationset-list-generator
spec:
  generators:
    - list:
        elements:
        - cluster: engineering-dev
          url: https://kubernetes.default.svc
        template:
          metadata: {}
          spec:
            destination: {}
            project: '{{cluster}}'
            source:
              path: applicationset/examples/template-override/{{cluster}}-override
              repoURL: https://github.com/argoproj/argo-cd.git
              targetRevision: HEAD
  template:
    metadata:
      name: '{{cluster}}-guestbook'
    spec:
      project: '{{cluster}}'
      source:
        path: applicationset/examples/template-override/default
        repoURL: https://github.com/argoproj/argo-cd.git
        targetRevision: HEAD
      destination:
        namespace: guestbook
        server: '{{url}}'
```

</details>

**Comment by oscrx** [comment]
LGTM

**Review by pdrastil — APPROVED** [review]

**Pull request merged**